exercicios/para-casa/ana-costa/BanksAccount.js

```diff
@@ -0,0 +1,104 @@
+import { Bank } from './Bank.js';
+import { Client } from './Client.js';
+
+export class BankAccount {
+  client;
+  bank;
+  accountNumber;
+  agencyNumber;
+  #balance = 0;
+
+
+  constructor(client,bank,accountNumber,agencyNumber){
+    if(!(client instanceof Client)) {
+      throw new Error('Não é possível abrir conta, cliente inválido');
+    }
+
+    if(!(bank instanceof Bank)) {
+      throw new Error('Não é possível abrir conta, cliente inválido');
+    }
+
+    this.client = client;
+    this.bank = bank;
+    this.accountNumber = accountNumber;
+    this.agencyNumber = agencyNumber;
+  }
+
+  credit(amount) {
+    if(typeof amount !== Number) {
+      throw new Error('Não é possível creditar valor, valor passado não é do tipo number');
+    }
+
+    if(amount < 0) {
+      throw new Error('Não é possível creditar valor, valor passado é menor que 0')
+    }
+
+    this.#balance += amount;
+
+    console.log(`O novo saldo da conta é: R$ ${this.#balance.toFixed(2)}`)
+  }
+
+  debit(amount) {
+    if(typeof amount !== Number) {
+      throw new Error('Não é possível debitar valor, valor passado não é do tipo number');
+    }
+
+    if(amount < 0) {
+      throw new Error('Não é possível debitar valor, valor passado é menor que 0')
+    }
+
+    this.#balance -= amount;
+
+    console.log(`O novo saldo da conta é: R$ ${this.#balance.toFixed(2)}`)
+  }
+
+   transferTo(anotherAccount, amount) {
+    //verifica se anotherAccount não é transferência de BankAccount, se não for, lança erro
+    if (!(anotherAccount instanceof BankAccount)) {
+      throw new Error('Não é possível fazer transferência, conta inválida');
+    }
+
+    //verifica se valor passado para transferência é menor do que 0, se sim, lança erro
+    if(amount < 0) {
+      throw new Error('Não é possível fazer transferência, valor passado é menor que 0');
+    }
+
+    //verifica se a transferência é para outro banco
+    isTransferToOtherBank = this.bank.bankCode !== anotherAccount.bank.bankCod;
+
+    // Verifica se isTransferToOtherBank é verdadeiro (a transferência é para outro banco).
+    // Se for verdadeiro, adiciona a taxa de transferência (this.bank.transferTax) ao valor original (amount) e armazena o resultado em totalAmount.
+    // Se for falso (a transferência não é para outro banco), apenas atribui o valor original (amount) a totalAmount.
+    totalAmount = isTransferToOtherBank ? (amount + this.bank.transferTax) : amount;
```

Comment:
Pelo que eu entendi do exercício era para somar 10% ao valor do amount. 
Desse jeito que você fez está somando só a taxa que é 0,01 ao amount. 
Eu fiz dessa forma: amount * (1 + this.bank.transferTax)